vllm/model_executor/layers/sampler.py

```diff
@@ -546,56 +586,97 @@ def _get_logprobs(
     logprobs: torch.Tensor,
     sampling_metadata: SamplingMetadata,
     sample_results: List[Tuple[List[int], List[int]]],
-) -> Tuple[List[Optional[List[Optional[Dict[int, float]]]]], List[List[Dict[
-        int, float]]]]:
-    # Prepare query indices
-    batched_logprobs_query_seq_indices: List[int] = []
-    batched_logprobs_query_token_indices: List[int] = []
+) -> Tuple[List[Optional[PromptLogprobs]], List[SampleLogprobs]]:
+    """Return sample lobprobs and prompt logprobs.
+
+    All metadata here has flattened query tokens in a batch. There could be one
+    or more query tokens per each request (seq_group).
+
+    Args:
+        logprobs: (num_query_tokens_across_batch, num_vocab). Each query token's
+            logprob per vocab.
+        sampling_metadata: The sampling metadata.
+        sample_results: (num_seq_groups) The tuple of (next_token_ids,
+            parent_ids) for each sequence group.
+
+    Returns:
+        A tuple of prompt and sample logprobs per sequence group in a batch.
+        If a seq_group doesn't require sample (e.g.,
+        sampling_metadata.do_samples[i] is False), the sampled logprobs can be
+        an empty string. If a seq group doesn't require prompt logprobs (e.g.,
+        if prompt_logprobs is None), the prompt logprobs may be None.
+    """
+    # The index of query token we care.
+    query_indices: List[int] = []
```

Comment:
Most of code change here is the rename of existing variables. Main logic is when do_sample is False, it doesn't compute logprobs